pkg/skaffold/jib/jib.go

```diff
@@ -78,3 +100,33 @@ func getDependencies(cmd *exec.Cmd) ([]string, error) {
 	sort.Strings(deps)
 	return deps, nil
 }
+
+// calculateRoots returns a list of possible symlink-expanded paths
+func calculateRoots(path string) ([]string, error) {
+	path, err := filepath.Abs(path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to resolve %s", path)
+	}
+	canonical, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to canonicalize workspace %s", path)
+	}
+	if path == canonical {
+		return []string{path}, nil
+	}
+	return []string{canonical, path}, nil
```

Comment:
Why are we returning both the symlink expanded and the non expanded version?

Reply:
This is @corneliusweig's [question](https://github.com/GoogleContainerTools/skaffold/pull/1926/files/4d799283170dde3fa008587382ae7a1a43a291fd#r272745633) too: basically Jib's `skaffold-files` goal just returns information provided from the Maven/Gradle.  These files *seem* to be absolute and canonical, but the docs don't actually guarantee that.  Expanding symlinks for *every* file seems like it could be expensive, so I instead expand the workspace so we can try comparing the expanded and unexpanded workspace paths as prefixes.